packages/block-library/src/separator/editor.scss

```diff
@@ -2,4 +2,10 @@
 	// Prevent margin collapsing so the area to select the separator is bigger.
 	padding-top: 0.1px;
 	padding-bottom: 0.1px;
+
+	// This is also set in style.scss, but needs a higher specificity in editor
+	// due to the way that color block supports adds additional background color styles.
+	&.wp-block-separator.is-style-dots {
+		background: none !important;
```

Comment:
I will take a closer look next week and see that is possible.

Reply:
@oandregal I looked into this and couldn't see a way around it as the color supports is already adding `! important` in other places:

<img width="271" alt="Screen Shot 2022-02-28 at 2 51 06 PM" src="https://user-images.githubusercontent.com/3629020/155911437-3bc72a70-c1db-45e1-8576-e94fa177c016.png">

The only potential option I could see is if we could remove the backGround support when `is-style-dots` is selected and just apply color instead, but this would remove the `has-background*` classes which could break backwards compat as all the existing blocks have both text and background classes applied, and themes may be relying on these.

Reply:
The issue seems to be that this block does things a bit differently depending on the kind of style variation applied. As I understand it, we don't need the automatic classes (`.has-<slug>-color`, etc) attached to the block wrapper in all the states of this block (when the dot style variation is applied).

How about using [__experimentalSkipSerialization](https://developer.wordpress.org/block-editor/explanations/architecture/styles/#current-limitations-of-the-block-supports-api) in the color block supports of the `block.json` and removing the `!important`s? I see that this PR is already doing some custom things in the `edit` and `save` functions, as well as the block stylesheet. When we do so, is because the normal default block support mechanism is not working as we want, so should look at disconnecting the automatic style serialization.

Reply:
> How about using [__experimentalSkipSerialization](https://developer.wordpress.org/block-editor/explanations/architecture/styles/#current-limitations-of-the-block-supports-api) in the color block supports of the block.json and removing the !importants?

It would be good to get this PR in: **Block Supports: Allow skipping serialization of individual features** https://github.com/WordPress/gutenberg/pull/36293.

That way we can choose specific block supports to "skip". Might mean less work, that not having to manually apply `attributes.styles`, later if we add block supports to the Separator.

Reply:
>The issue seems to be that this block does things a bit differently depending on the kind of style variation applied. 
>As I understand it, we don't need the automatic classes (.has-<slug>-color, etc) attached to the block wrapper in 
>all the states of this block (when the dot style variation is applied).

@oandregal yes, ideally the `.has-<slug>-color` classes should just be applied when the block is in the relevant state, eg. `background-color` when it is standard  and `color` when it is dotted, in which case skipping serialization would make sense.

Unfortunately, the block currently applies both classes in both instances, and we have no idea how theme authors might be relying on this current combination of selectors. This means that moving to using the single selector in the specific instances would be a breaking change. I am not sure that the value gained from this change would be worth the potential impact on themes?

Also, the `background-color: none ! important`  is still going to be needed in order to make sure that theme `background-color` is overridden when the dotted style is selected [as noted here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/separator/style.scss#L10). This has always been included in the block,  this additional setting was only added to make sure it was still applied in the editor with the addition of the new color supports settings.

For the above reasons I think we should leave the classes are they currently are and leave the `background-color: none ! important`  in place, but open suggestions for ways that we can remove it while avoiding a breaking change, or also open to feedback on whether this change is important enough to warrant a breaking change.

Reply:
This is a hard one. 

The background classes don't make sense on the dots style, but yeah... they're there ,and some folks might be counting on that. Avoiding pain for users and theme developers seems like it should take priority.

Could we deprecate and migrate in a follow up PR?

On a higher level, we've chosen, for whatever reasons, to use `!important` in various places so we probably have to deal with it as best as we can until we find a way to mitigate such situations.

Reply:
> re: having classes pertinent to the state.

Yeah, I understand the concerns you've brought up: that change alone would need wider testing in themes. It's fine by me to leave it as it is in this PR and explore its removal in a follow up.

Regarding the other thing I've mentioned: I still think we need to use `__experimentalSkipSerialization` in the `color` support since we're already managing the serialization by the block.

Reply:
> Regarding the other thing I've mentioned: I still think we need to use `__experimentalSkipSerialization` in the `color` support since we're already managing the serialization by the block.

Good point @oandregal, doing that makes it a bit clearer that this isn't a standard use of the color supports. I have made that change now,